### PR TITLE
Upd linkbot

### PR DESCRIPTION
**Pull request by oleorhagen**

### CODE REVIEW

File: scripts/linkbot/monitor-ub-server.js [hunk 1]
```diff
@@ -28,27 +28,42 @@ JSDOM.fromURL(url, {}).then(dom => {
...
     })
-});
+        .catch(err => {
+            console.log('Failed to get the update URL');
+            console.log(`${url}${releaseVersion}/release/`);
+            JSDOM.fromURL(`${url}${releasedVersion}/beta/`, {}).then(dom => {
+                console.log('Only the beta is out still');
+                process.exit(0);
+            });
+            process.exit(1);
```
Comment (mzedel) [inline scripts/linkbot/monitor-ub-server.js:61]: This I could imagine to cause problems... as the process would exit with 1 once the `fromURL` call was issued, and then the `exit(0)` might never reach anyone...
I think an option to get this done might be:
```
let exitCode = 0;
JSDOM.fromURL(...)
  .then(dom => {
    ...
    return Promise.resolve();
  })
  .catch(err => {
    ...
    exitCode = 1;
    return JSDOM.fromURL(...)
  })
  .then(dom => {
    ...
    exitCode = 0;
  })
  .finally(() => process.exit(exitCode));
```

**Review by mzedel — COMMENTED** [review]
Good to see you sticking to a nice language 😄

File: scripts/linkbot/monitor-ub-server.js [hunk 2]
```diff
@@ -28,27 +28,42 @@ JSDOM.fromURL(url, {}).then(dom => {
...
-            console.log(`Ubuntu server image has a new release: ${match}`)
-            updateURLLink(`${target}=\"${match}\"`, target)
+            console.log(`Ubuntu server image has a new release: ${match}`);
+            updateURLLink(`${target}=\"${match}\"`, target);
         }
     })
-});
+        .catch(err => {
+            console.log('Failed to get the update URL');
+            console.log(`${url}${releaseVersion}/release/`);
```
[the remark this answers is not in the document]
Comment (oleorhagen) [reply scripts/linkbot/monitor-ub-server.js:56]: Thx, sloppy

File: scripts/linkbot/monitor-ub-server.js [hunk 3]
```diff
@@ -28,27 +28,42 @@ JSDOM.fromURL(url, {}).then(dom => {
...
+            JSDOM.fromURL(`${url}${releasedVersion}/beta/`, {}).then(dom => {
+                console.log('Only the beta is out still');
+                process.exit(0);
+            });
+            process.exit(1);
+        });
+})
+    .catch(err => {
+        console.log('Failed to get the URL');
+        console.log(`${url}`);
```
[the remark this answers is not in the document]
Comment (oleorhagen) [reply scripts/linkbot/monitor-ub-server.js:66]: Much easier :smile_cat:

File: scripts/linkbot/monitor-ub-server.js (hunk 1 above)
Comment (oleorhagen) [reply scripts/linkbot/monitor-ub-server.js:61]: Could another idea here be to remove the `exits`, and simply catch the ones I want to not fail, and re-raise in the ones I want to fail ?

Comment (mzedel) [reply scripts/linkbot/monitor-ub-server.js:61]: absolutely - just throw whenever and make the crash exit the process 👍

Comment (oleorhagen) [reply scripts/linkbot/monitor-ub-server.js]: Done :)

File: scripts/linkbot/monitor-ub-server.js [hunk 4]
```diff
@@ -28,27 +28,41 @@ JSDOM.fromURL(url, {}).then(dom => {
...
+                    console.log('Only the beta is out still');
+                });
+        });
+})
+    .catch(err => {
+        console.log('Failed to get the URL');
+        console.log(url);
+        console.log(err);
+        throw err;
+    });
```
Comment (mzedel) [inline scripts/linkbot/monitor-ub-server.js:68]: This should work, but it's not very pretty IMO... you could return each `JSDOM.fromURL` call to get rid of the nesting and help readability, but I'm fine with merging it as is... "additional changes are left as a reader exercise" or something along those lines 😄

**Review by mzedel — APPROVED** [review]

**Pull request merged**